### PR TITLE
Update to TRestDetectorReadout

### PR DESCRIPTION
**Pull request by lobis**
lobis Large: 510 [](https://github.com/rest-for-physics/detectorlib/commits/lobis-refactoring)

While developing a feature request I realised the readout classes have not been worked on for a while and would benefit from some review.

The changes submitted by this PR should not alter the functionality of the classes but instead make the code more consisten with other more recently developed REST-for-Physics code and (implicit) style guidelines.

### CODE REVIEW

File: inc/TRestDetectorReadoutModule.h [hunk 1]
```diff
@@ -34,29 +33,23 @@
...
    private:
     Int_t fModuleID;  ///< The module id given by the readout definition.
 
     TString fModuleName;  ///< The assigned module name.
 
-    Double_t fModuleOriginX;  ///< The module x-position (left-bottom corner)
-                              ///< relative to the readout plane position.
-    Double_t fModuleOriginY;  ///< The module y-position (left-bottom corner)
-                              ///< relative to the readout plane position.
+    TVector2 fModuleOrigin;  ///< The module (x, y) position relative to the readout plane position.
```
[the remark this answers is not in the document]
Comment (lobis) [reply inc/TRestDetectorReadoutModule.h:42]: Unfortunately I think I broke something even though I was careful and it probably too hard to fix now (too many changes).

[the remark this answers is not in the document]
Comment (jgalan) [reply inc/TRestDetectorReadoutModule.h:42]: I am regenerating the readout we use for validation. The definition can be found at this repository: https://lfna.unizar.es/gifna/trexdm-readouts

[the remark this answers is not in the document]
Comment (jgalan) [reply inc/TRestDetectorReadoutModule.h:42]: Apparently schema evolution is not clever enough

[the remark this answers is not in the document]
Comment (jgalan) [reply inc/TRestDetectorReadoutModule.h:42]: > too many changes

So better to do smaller commits and check step by step

**Comment by jgalan** [comment]
I was trying to work out a solution. I think the PR should be split. For instance, I would place the `TObject` removal and the data member changes towards `TVector2` later on in a separate PR.

**Comment by jgalan** [comment]
In reality the classes do not need to inherit from `TRestMetadata`, but the idea behind their inheritance was a better implementation such as:

```
<TRestDetectorReadout>
      <TRestDetectorModule>
        ....
```

So that we can also retrieve the children as a stand-alone metadata structure, using `TRestMetadata::InstantiateChildMetadata`.

**Comment by lobis** [comment]
Closing in favour of https://github.com/rest-for-physics/detectorlib/pull/86

**Pull request closed without merging**